File: tyrant/python/update/tyrant_update.py
#! /usr/bin/python3
# -*- coding: utf-8 -*-
#
#	tyrant_update.py
#
#					Feb/06/2015
#
import	sys
#
import memcache
#
sys.path.append ('/var/www/data_base/common/python_common')
from mcache_manipulate import mcache_update_proc
# ------------------------------------------------------------
print ("*** 開始 ***")
#
key_in = sys.argv[1]
population_in = int (sys.argv[2])
print ("%s\t%d" % (key_in, population_in))
#
mc = memcache.Client(['ubuntu_1504:1978'])
#
mcache_update_proc (mc,key_in,population_in)
#
print ("*** 終了 ***")
#
# ------------------------------------------------------------

File: tyrant/node/update/tyrant_update.js
#! /usr/bin/nodejs
// -------------------------------------------------------------
//	tyrant_update.js
//
//					Feb/15/2016
// -------------------------------------------------------------
var memcache = require('memcache')
var node_common = '/var/www/data_base/common/node_common'

var text_manipulate=require (node_common + '/text_manipulate')
var json_manipulate=require (node_common + '/json_manipulate')

// -------------------------------------------------------------
function set_proc (client,key,value)
{
	client.set  (key,value,function (error,result){ })
}

// -------------------------------------------------------------
function update_proc (client,key,population)
{
	client.get (key,function(error,reply)
		{
		if (!error)
			{
			if (reply != null)
				{
				var json_out = json_manipulate.json_update_proc (reply,population)
				set_proc  (client,key,json_out)
				}
			}
		client.close()
		console.log("*** 終了 ***")
		})
}

// -------------------------------------------------------------
var onConnect = function()
{

	update_proc (client,key_in,population_in)

}

// -------------------------------------------------------------
console.log("*** 開始 ***")
var key_in=process.argv[2]
var population_in=process.argv[3]

console.log (key_in + "\t" + population_in)

var client = new memcache.Client (1978,'host_ubuntu')
client.connect()
client.addHandler(onConnect)

// -------------------------------------------------------------

